<html><body>
<style>

body, h1, h2, h3, div, span, p, pre, a {
  margin: 0;
  padding: 0;
  border: 0;
  font-weight: inherit;
  font-style: inherit;
  font-size: 100%;
  font-family: inherit;
  vertical-align: baseline;
}

body {
  font-size: 13px;
  padding: 1em;
}

h1 {
  font-size: 26px;
  margin-bottom: 1em;
}

h2 {
  font-size: 24px;
  margin-bottom: 1em;
}

h3 {
  font-size: 20px;
  margin-bottom: 1em;
  margin-top: 1em;
}

pre, code {
  line-height: 1.5;
  font-family: Monaco, 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', 'Lucida Console', monospace;
}

pre {
  margin-top: 0.5em;
}

h1, h2, h3, p {
  font-family: Arial, sans serif;
}

h1, h2, h3 {
  border-bottom: solid #CCC 1px;
}

.toc_element {
  margin-top: 0.5em;
}

.firstline {
  margin-left: 2 em;
}

.method  {
  margin-top: 1em;
  border: solid 1px #CCC;
  padding: 1em;
  background: #EEE;
}

.details {
  font-weight: bold;
  font-size: 14px;
}

</style>

<h1><a href="genomics_v1beta.html">Genomics API</a> . <a href="genomics_v1beta.variants.html">variants</a></h1>
<h2>Instance Methods</h2>
<p class="toc_element">
  <code><a href="#create">create(body)</a></code></p>
<p class="firstline">Creates a new variant.</p>
<p class="toc_element">
  <code><a href="#delete">delete(variantId)</a></code></p>
<p class="firstline">Deletes a variant.</p>
<p class="toc_element">
  <code><a href="#export">export(body)</a></code></p>
<p class="firstline">Exports variant data to an external destination.</p>
<p class="toc_element">
  <code><a href="#get">get(variantId)</a></code></p>
<p class="firstline">Gets a variant by ID.</p>
<p class="toc_element">
  <code><a href="#getSummary">getSummary(datasetId=None)</a></code></p>
<p class="firstline">Gets a summary of all the variant data in a dataset.</p>
<p class="toc_element">
  <code><a href="#import_">import_(body)</a></code></p>
<p class="firstline">Creates variant data by asynchronously importing the provided information.</p>
<p class="toc_element">
  <code><a href="#patch">patch(variantId, body)</a></code></p>
<p class="firstline">Updates a variant. This method supports patch semantics.</p>
<p class="toc_element">
  <code><a href="#search">search(body)</a></code></p>
<p class="firstline">Gets a list of variants matching the criteria.</p>
<p class="toc_element">
  <code><a href="#update">update(variantId, body)</a></code></p>
<p class="firstline">Updates a variant.</p>
<h3>Method Details</h3>
<div class="method">
    <code class="details" id="create">create(body)</code>
  <pre>Creates a new variant.

Args:
  body: object, The request body. (required)
    The object takes the form of:

{ # A Variant represents a change in DNA sequence relative to some reference. For example, a Variant could represent a SNP or an insertion. Variants belong to a Dataset.
  "info": { # A map of additional variant information.
    "a_key": [ # A string which maps to an array of values.
      "A String",
    ],
  },
  "end": "A String", # The end position (1-based) of this variant. This corresponds to the first base after the last base in the reference allele. So, the length of the reference allele is (end - position). This is useful for variants that don't explicitly give alternate bases, for example large deletions.
  "calls": [ # The variant calls for this particular variant. Each one represents the determination of genotype with respect to this variant.
    { # A Call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a Call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a callset with the name NA12345.
      "info": { # A map of additional variant call information.
        "a_key": [ # A string which maps to an array of values.
          "A String",
        ],
      },
      "genotype": [ # The genotype of this variant call. Each value represents either the value of the referenceBases field or a 1-based index into alternateBases. If a variant had a referenceBases field of "T" and an alternateBases value of ["A", "C"], and the genotype was [2, 1], that would mean the call represented the heterozygous value "CA" for this variant. If the genotype was instead [0, 1], the represented value would be "TA". Ordering of the genotype values is important if the phaseset field is present. If a genotype is not called (that is, a "." is present in the GT string) -1 is returned.
        42,
      ],
      "callsetId": "A String", # The ID of the callset this variant call belongs to.
      "phaseset": "A String", # If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls on the same contig which have the same phaseset value. When importing data from VCF, if the genotype data was phased but no phase set was specified this field will be set to "*".
      "genotypeLikelihood": [ # The genotype likelihoods for this variant call. Each array entry represents how likely a specific genotype is for this call. The value ordering is defined by the GL tag in the VCF spec.
        3.14,
      ],
      "callsetName": "A String", # The name of the callset this variant call belongs to.
    },
  ],
  "created": "A String", # The date this variant was created, in milliseconds from the epoch.
  "referenceBases": "A String", # The reference bases for this variant. They start at the given position.
  "names": [ # Names for the variant, for example a RefSNP ID.
    "A String",
  ],
  "alternateBases": [ # The bases that appear instead of the reference bases.
    "A String",
  ],
  "position": "A String", # The position at which this variant occurs (1-based). This corresponds to the first base of the string of reference bases.
  "contig": "A String", # The contig on which this variant occurs. (such as 'chr20' or 'X')
  "id": "A String", # The Google generated ID of the variant, immutable.
  "datasetId": "A String", # The ID of the dataset this variant belongs to.
}


Returns:
  An object of the form:

    { # A Variant represents a change in DNA sequence relative to some reference. For example, a Variant could represent a SNP or an insertion. Variants belong to a Dataset.
    "info": { # A map of additional variant information.
      "a_key": [ # A string which maps to an array of values.
        "A String",
      ],
    },
    "end": "A String", # The end position (1-based) of this variant. This corresponds to the first base after the last base in the reference allele. So, the length of the reference allele is (end - position). This is useful for variants that don't explicitly give alternate bases, for example large deletions.
    "calls": [ # The variant calls for this particular variant. Each one represents the determination of genotype with respect to this variant.
      { # A Call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a Call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a callset with the name NA12345.
        "info": { # A map of additional variant call information.
          "a_key": [ # A string which maps to an array of values.
            "A String",
          ],
        },
        "genotype": [ # The genotype of this variant call. Each value represents either the value of the referenceBases field or a 1-based index into alternateBases. If a variant had a referenceBases field of "T" and an alternateBases value of ["A", "C"], and the genotype was [2, 1], that would mean the call represented the heterozygous value "CA" for this variant. If the genotype was instead [0, 1], the represented value would be "TA". Ordering of the genotype values is important if the phaseset field is present. If a genotype is not called (that is, a "." is present in the GT string) -1 is returned.
          42,
        ],
        "callsetId": "A String", # The ID of the callset this variant call belongs to.
        "phaseset": "A String", # If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls on the same contig which have the same phaseset value. When importing data from VCF, if the genotype data was phased but no phase set was specified this field will be set to "*".
        "genotypeLikelihood": [ # The genotype likelihoods for this variant call. Each array entry represents how likely a specific genotype is for this call. The value ordering is defined by the GL tag in the VCF spec.
          3.14,
        ],
        "callsetName": "A String", # The name of the callset this variant call belongs to.
      },
    ],
    "created": "A String", # The date this variant was created, in milliseconds from the epoch.
    "referenceBases": "A String", # The reference bases for this variant. They start at the given position.
    "names": [ # Names for the variant, for example a RefSNP ID.
      "A String",
    ],
    "alternateBases": [ # The bases that appear instead of the reference bases.
      "A String",
    ],
    "position": "A String", # The position at which this variant occurs (1-based). This corresponds to the first base of the string of reference bases.
    "contig": "A String", # The contig on which this variant occurs. (such as 'chr20' or 'X')
    "id": "A String", # The Google generated ID of the variant, immutable.
    "datasetId": "A String", # The ID of the dataset this variant belongs to.
  }</pre>
</div>

<div class="method">
    <code class="details" id="delete">delete(variantId)</code>
  <pre>Deletes a variant.

Args:
  variantId: string, The ID of the variant to be deleted. (required)
</pre>
</div>

<div class="method">
    <code class="details" id="export">export(body)</code>
  <pre>Exports variant data to an external destination.

Args:
  body: object, The request body. (required)
    The object takes the form of:

{ # The variant data export request.
    "format": "A String", # The format for the exported data.
    "projectId": "A String", # The Google Cloud project number that owns this export. This is the project that will be billed.
    "callsetIds": [ # If provided, only variant call information from the specified callsets will be exported. By default all variant calls are exported.
      "A String",
    ],
    "bigqueryDataset": "A String", # The BigQuery dataset to export data to. Note that this is distinct from the Genomics concept of "dataset". The caller must have WRITE access to this BigQuery dataset.
    "datasetId": "A String", # Required. The ID of the dataset that contains variant data which should be exported. The caller must have READ access to this dataset.
    "bigqueryTable": "A String", # The BigQuery table to export data to. The caller must have WRITE access to this BigQuery table.
  }


Returns:
  An object of the form:

    { # The variant data export response.
    "jobId": "A String", # A job ID that can be used to get status information.
  }</pre>
</div>

<div class="method">
    <code class="details" id="get">get(variantId)</code>
  <pre>Gets a variant by ID.

Args:
  variantId: string, The ID of the variant. (required)

Returns:
  An object of the form:

    { # A Variant represents a change in DNA sequence relative to some reference. For example, a Variant could represent a SNP or an insertion. Variants belong to a Dataset.
    "info": { # A map of additional variant information.
      "a_key": [ # A string which maps to an array of values.
        "A String",
      ],
    },
    "end": "A String", # The end position (1-based) of this variant. This corresponds to the first base after the last base in the reference allele. So, the length of the reference allele is (end - position). This is useful for variants that don't explicitly give alternate bases, for example large deletions.
    "calls": [ # The variant calls for this particular variant. Each one represents the determination of genotype with respect to this variant.
      { # A Call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a Call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a callset with the name NA12345.
        "info": { # A map of additional variant call information.
          "a_key": [ # A string which maps to an array of values.
            "A String",
          ],
        },
        "genotype": [ # The genotype of this variant call. Each value represents either the value of the referenceBases field or a 1-based index into alternateBases. If a variant had a referenceBases field of "T" and an alternateBases value of ["A", "C"], and the genotype was [2, 1], that would mean the call represented the heterozygous value "CA" for this variant. If the genotype was instead [0, 1], the represented value would be "TA". Ordering of the genotype values is important if the phaseset field is present. If a genotype is not called (that is, a "." is present in the GT string) -1 is returned.
          42,
        ],
        "callsetId": "A String", # The ID of the callset this variant call belongs to.
        "phaseset": "A String", # If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls on the same contig which have the same phaseset value. When importing data from VCF, if the genotype data was phased but no phase set was specified this field will be set to "*".
        "genotypeLikelihood": [ # The genotype likelihoods for this variant call. Each array entry represents how likely a specific genotype is for this call. The value ordering is defined by the GL tag in the VCF spec.
          3.14,
        ],
        "callsetName": "A String", # The name of the callset this variant call belongs to.
      },
    ],
    "created": "A String", # The date this variant was created, in milliseconds from the epoch.
    "referenceBases": "A String", # The reference bases for this variant. They start at the given position.
    "names": [ # Names for the variant, for example a RefSNP ID.
      "A String",
    ],
    "alternateBases": [ # The bases that appear instead of the reference bases.
      "A String",
    ],
    "position": "A String", # The position at which this variant occurs (1-based). This corresponds to the first base of the string of reference bases.
    "contig": "A String", # The contig on which this variant occurs. (such as 'chr20' or 'X')
    "id": "A String", # The Google generated ID of the variant, immutable.
    "datasetId": "A String", # The ID of the dataset this variant belongs to.
  }</pre>
</div>

<div class="method">
    <code class="details" id="getSummary">getSummary(datasetId=None)</code>
  <pre>Gets a summary of all the variant data in a dataset.

Args:
  datasetId: string, Required. The ID of the dataset to get variant summary information for.

Returns:
  An object of the form:

    { # The variants summary response.
    "contigBounds": [ # A list of all contigs used by the variants in a dataset with associated coordinate upper bounds for each one.
      { # ContigBound records an upper bound for the starting coordinate of variants in a particular contig.
        "contig": "A String", # The contig the bound is associate with.
        "upperBound": "A String", # An upper bound (inclusive) on the starting coordinate of any variant in the contig.
      },
    ],
    "metadata": [ # The metadata associated with this dataset.
      { # Metadata describes a single piece of variant call metadata. These data include a top level key and either a single value string (value) or a list of key-value pairs (info.) Value and info are mutually exclusive.
        "info": { # Remaining structured metadata key-value pairs.
          "a_key": [ # A string which maps to an array of values.
            "A String",
          ],
        },
        "description": "A String", # A textual description of this metadata.
        "number": "A String", # The number of values that can be included in a field described by this metadata.
        "value": "A String", # The value field for simple metadata
        "key": "A String", # The top-level key.
        "type": "A String", # The type of data. Possible types include: Integer, Float, Flag, Character, and String.
        "id": "A String", # User-provided ID field, not enforced by this API. Two or more pieces of structured metadata with identical id and key fields are considered equivalent.
      },
    ],
  }</pre>
</div>

<div class="method">
    <code class="details" id="import_">import_(body)</code>
  <pre>Creates variant data by asynchronously importing the provided information.

Args:
  body: object, The request body. (required)
    The object takes the form of:

{ # The variant data import request.
    "sourceUris": [ # A list of URIs pointing at VCF files in Google Cloud Storage. See the VCF Specification for more details on the input format.
      "A String",
    ],
    "datasetId": "A String", # Required. The dataset to which variant data should be imported.
  }


Returns:
  An object of the form:

    { # The variant data import response.
    "jobId": "A String", # A job ID that can be used to get status information.
  }</pre>
</div>

<div class="method">
    <code class="details" id="patch">patch(variantId, body)</code>
  <pre>Updates a variant. This method supports patch semantics.

Args:
  variantId: string, The ID of the variant to be updated.. (required)
  body: object, The request body. (required)
    The object takes the form of:

{ # A Variant represents a change in DNA sequence relative to some reference. For example, a Variant could represent a SNP or an insertion. Variants belong to a Dataset.
  "info": { # A map of additional variant information.
    "a_key": [ # A string which maps to an array of values.
      "A String",
    ],
  },
  "end": "A String", # The end position (1-based) of this variant. This corresponds to the first base after the last base in the reference allele. So, the length of the reference allele is (end - position). This is useful for variants that don't explicitly give alternate bases, for example large deletions.
  "calls": [ # The variant calls for this particular variant. Each one represents the determination of genotype with respect to this variant.
    { # A Call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a Call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a callset with the name NA12345.
      "info": { # A map of additional variant call information.
        "a_key": [ # A string which maps to an array of values.
          "A String",
        ],
      },
      "genotype": [ # The genotype of this variant call. Each value represents either the value of the referenceBases field or a 1-based index into alternateBases. If a variant had a referenceBases field of "T" and an alternateBases value of ["A", "C"], and the genotype was [2, 1], that would mean the call represented the heterozygous value "CA" for this variant. If the genotype was instead [0, 1], the represented value would be "TA". Ordering of the genotype values is important if the phaseset field is present. If a genotype is not called (that is, a "." is present in the GT string) -1 is returned.
        42,
      ],
      "callsetId": "A String", # The ID of the callset this variant call belongs to.
      "phaseset": "A String", # If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls on the same contig which have the same phaseset value. When importing data from VCF, if the genotype data was phased but no phase set was specified this field will be set to "*".
      "genotypeLikelihood": [ # The genotype likelihoods for this variant call. Each array entry represents how likely a specific genotype is for this call. The value ordering is defined by the GL tag in the VCF spec.
        3.14,
      ],
      "callsetName": "A String", # The name of the callset this variant call belongs to.
    },
  ],
  "created": "A String", # The date this variant was created, in milliseconds from the epoch.
  "referenceBases": "A String", # The reference bases for this variant. They start at the given position.
  "names": [ # Names for the variant, for example a RefSNP ID.
    "A String",
  ],
  "alternateBases": [ # The bases that appear instead of the reference bases.
    "A String",
  ],
  "position": "A String", # The position at which this variant occurs (1-based). This corresponds to the first base of the string of reference bases.
  "contig": "A String", # The contig on which this variant occurs. (such as 'chr20' or 'X')
  "id": "A String", # The Google generated ID of the variant, immutable.
  "datasetId": "A String", # The ID of the dataset this variant belongs to.
}


Returns:
  An object of the form:

    { # A Variant represents a change in DNA sequence relative to some reference. For example, a Variant could represent a SNP or an insertion. Variants belong to a Dataset.
    "info": { # A map of additional variant information.
      "a_key": [ # A string which maps to an array of values.
        "A String",
      ],
    },
    "end": "A String", # The end position (1-based) of this variant. This corresponds to the first base after the last base in the reference allele. So, the length of the reference allele is (end - position). This is useful for variants that don't explicitly give alternate bases, for example large deletions.
    "calls": [ # The variant calls for this particular variant. Each one represents the determination of genotype with respect to this variant.
      { # A Call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a Call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a callset with the name NA12345.
        "info": { # A map of additional variant call information.
          "a_key": [ # A string which maps to an array of values.
            "A String",
          ],
        },
        "genotype": [ # The genotype of this variant call. Each value represents either the value of the referenceBases field or a 1-based index into alternateBases. If a variant had a referenceBases field of "T" and an alternateBases value of ["A", "C"], and the genotype was [2, 1], that would mean the call represented the heterozygous value "CA" for this variant. If the genotype was instead [0, 1], the represented value would be "TA". Ordering of the genotype values is important if the phaseset field is present. If a genotype is not called (that is, a "." is present in the GT string) -1 is returned.
          42,
        ],
        "callsetId": "A String", # The ID of the callset this variant call belongs to.
        "phaseset": "A String", # If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls on the same contig which have the same phaseset value. When importing data from VCF, if the genotype data was phased but no phase set was specified this field will be set to "*".
        "genotypeLikelihood": [ # The genotype likelihoods for this variant call. Each array entry represents how likely a specific genotype is for this call. The value ordering is defined by the GL tag in the VCF spec.
          3.14,
        ],
        "callsetName": "A String", # The name of the callset this variant call belongs to.
      },
    ],
    "created": "A String", # The date this variant was created, in milliseconds from the epoch.
    "referenceBases": "A String", # The reference bases for this variant. They start at the given position.
    "names": [ # Names for the variant, for example a RefSNP ID.
      "A String",
    ],
    "alternateBases": [ # The bases that appear instead of the reference bases.
      "A String",
    ],
    "position": "A String", # The position at which this variant occurs (1-based). This corresponds to the first base of the string of reference bases.
    "contig": "A String", # The contig on which this variant occurs. (such as 'chr20' or 'X')
    "id": "A String", # The Google generated ID of the variant, immutable.
    "datasetId": "A String", # The ID of the dataset this variant belongs to.
  }</pre>
</div>

<div class="method">
    <code class="details" id="search">search(body)</code>
  <pre>Gets a list of variants matching the criteria.

Args:
  body: object, The request body. (required)
    The object takes the form of:

{ # The variant search request.
    "endPosition": "A String", # Required. The end of the window (1-based, inclusive) for which overlapping variants should be returned.
    "startPosition": "A String", # Required. The beginning of the window (1-based, inclusive) for which overlapping variants should be returned.
    "callsetNames": [ # Only return variant calls which belong to callsets which have exactly these names. Leaving this blank returns all variant calls. At most one of callsetNames or callsetIds should be provided.
      "A String",
    ],
    "maxResults": "A String", # The maximum number of variants to return.
    "pageToken": "A String", # The continuation token, which is used to page through large result sets. To get the next page of results, set this parameter to the value of "nextPageToken" from the previous response.
    "callsetIds": [ # Only return variant calls which belong to callsets with these ids. Leaving this blank returns all variant calls. At most one of callsetNames or callsetIds should be provided.
      "A String",
    ],
    "variantName": "A String", # Only return variants which have exactly this name.
    "contig": "A String", # Required. Only return variants on this contig.
    "datasetId": "A String", # Required. The ID of the dataset to search.
  }


Returns:
  An object of the form:

    { # The variant search response.
    "nextPageToken": "A String", # The continuation token, which is used to page through large result sets. Provide this value in a subsequent request to return the next page of results. This field will be empty if there aren't any additional results.
    "variants": [ # The list of matching Variants.
      { # A Variant represents a change in DNA sequence relative to some reference. For example, a Variant could represent a SNP or an insertion. Variants belong to a Dataset.
        "info": { # A map of additional variant information.
          "a_key": [ # A string which maps to an array of values.
            "A String",
          ],
        },
        "end": "A String", # The end position (1-based) of this variant. This corresponds to the first base after the last base in the reference allele. So, the length of the reference allele is (end - position). This is useful for variants that don't explicitly give alternate bases, for example large deletions.
        "calls": [ # The variant calls for this particular variant. Each one represents the determination of genotype with respect to this variant.
          { # A Call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a Call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a callset with the name NA12345.
            "info": { # A map of additional variant call information.
              "a_key": [ # A string which maps to an array of values.
                "A String",
              ],
            },
            "genotype": [ # The genotype of this variant call. Each value represents either the value of the referenceBases field or a 1-based index into alternateBases. If a variant had a referenceBases field of "T" and an alternateBases value of ["A", "C"], and the genotype was [2, 1], that would mean the call represented the heterozygous value "CA" for this variant. If the genotype was instead [0, 1], the represented value would be "TA". Ordering of the genotype values is important if the phaseset field is present. If a genotype is not called (that is, a "." is present in the GT string) -1 is returned.
              42,
            ],
            "callsetId": "A String", # The ID of the callset this variant call belongs to.
            "phaseset": "A String", # If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls on the same contig which have the same phaseset value. When importing data from VCF, if the genotype data was phased but no phase set was specified this field will be set to "*".
            "genotypeLikelihood": [ # The genotype likelihoods for this variant call. Each array entry represents how likely a specific genotype is for this call. The value ordering is defined by the GL tag in the VCF spec.
              3.14,
            ],
            "callsetName": "A String", # The name of the callset this variant call belongs to.
          },
        ],
        "created": "A String", # The date this variant was created, in milliseconds from the epoch.
        "referenceBases": "A String", # The reference bases for this variant. They start at the given position.
        "names": [ # Names for the variant, for example a RefSNP ID.
          "A String",
        ],
        "alternateBases": [ # The bases that appear instead of the reference bases.
          "A String",
        ],
        "position": "A String", # The position at which this variant occurs (1-based). This corresponds to the first base of the string of reference bases.
        "contig": "A String", # The contig on which this variant occurs. (such as 'chr20' or 'X')
        "id": "A String", # The Google generated ID of the variant, immutable.
        "datasetId": "A String", # The ID of the dataset this variant belongs to.
      },
    ],
  }</pre>
</div>

<div class="method">
    <code class="details" id="update">update(variantId, body)</code>
  <pre>Updates a variant.

Args:
  variantId: string, The ID of the variant to be updated.. (required)
  body: object, The request body. (required)
    The object takes the form of:

{ # A Variant represents a change in DNA sequence relative to some reference. For example, a Variant could represent a SNP or an insertion. Variants belong to a Dataset.
  "info": { # A map of additional variant information.
    "a_key": [ # A string which maps to an array of values.
      "A String",
    ],
  },
  "end": "A String", # The end position (1-based) of this variant. This corresponds to the first base after the last base in the reference allele. So, the length of the reference allele is (end - position). This is useful for variants that don't explicitly give alternate bases, for example large deletions.
  "calls": [ # The variant calls for this particular variant. Each one represents the determination of genotype with respect to this variant.
    { # A Call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a Call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a callset with the name NA12345.
      "info": { # A map of additional variant call information.
        "a_key": [ # A string which maps to an array of values.
          "A String",
        ],
      },
      "genotype": [ # The genotype of this variant call. Each value represents either the value of the referenceBases field or a 1-based index into alternateBases. If a variant had a referenceBases field of "T" and an alternateBases value of ["A", "C"], and the genotype was [2, 1], that would mean the call represented the heterozygous value "CA" for this variant. If the genotype was instead [0, 1], the represented value would be "TA". Ordering of the genotype values is important if the phaseset field is present. If a genotype is not called (that is, a "." is present in the GT string) -1 is returned.
        42,
      ],
      "callsetId": "A String", # The ID of the callset this variant call belongs to.
      "phaseset": "A String", # If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls on the same contig which have the same phaseset value. When importing data from VCF, if the genotype data was phased but no phase set was specified this field will be set to "*".
      "genotypeLikelihood": [ # The genotype likelihoods for this variant call. Each array entry represents how likely a specific genotype is for this call. The value ordering is defined by the GL tag in the VCF spec.
        3.14,
      ],
      "callsetName": "A String", # The name of the callset this variant call belongs to.
    },
  ],
  "created": "A String", # The date this variant was created, in milliseconds from the epoch.
  "referenceBases": "A String", # The reference bases for this variant. They start at the given position.
  "names": [ # Names for the variant, for example a RefSNP ID.
    "A String",
  ],
  "alternateBases": [ # The bases that appear instead of the reference bases.
    "A String",
  ],
  "position": "A String", # The position at which this variant occurs (1-based). This corresponds to the first base of the string of reference bases.
  "contig": "A String", # The contig on which this variant occurs. (such as 'chr20' or 'X')
  "id": "A String", # The Google generated ID of the variant, immutable.
  "datasetId": "A String", # The ID of the dataset this variant belongs to.
}


Returns:
  An object of the form:

    { # A Variant represents a change in DNA sequence relative to some reference. For example, a Variant could represent a SNP or an insertion. Variants belong to a Dataset.
    "info": { # A map of additional variant information.
      "a_key": [ # A string which maps to an array of values.
        "A String",
      ],
    },
    "end": "A String", # The end position (1-based) of this variant. This corresponds to the first base after the last base in the reference allele. So, the length of the reference allele is (end - position). This is useful for variants that don't explicitly give alternate bases, for example large deletions.
    "calls": [ # The variant calls for this particular variant. Each one represents the determination of genotype with respect to this variant.
      { # A Call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a Call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a callset with the name NA12345.
        "info": { # A map of additional variant call information.
          "a_key": [ # A string which maps to an array of values.
            "A String",
          ],
        },
        "genotype": [ # The genotype of this variant call. Each value represents either the value of the referenceBases field or a 1-based index into alternateBases. If a variant had a referenceBases field of "T" and an alternateBases value of ["A", "C"], and the genotype was [2, 1], that would mean the call represented the heterozygous value "CA" for this variant. If the genotype was instead [0, 1], the represented value would be "TA". Ordering of the genotype values is important if the phaseset field is present. If a genotype is not called (that is, a "." is present in the GT string) -1 is returned.
          42,
        ],
        "callsetId": "A String", # The ID of the callset this variant call belongs to.
        "phaseset": "A String", # If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls on the same contig which have the same phaseset value. When importing data from VCF, if the genotype data was phased but no phase set was specified this field will be set to "*".
        "genotypeLikelihood": [ # The genotype likelihoods for this variant call. Each array entry represents how likely a specific genotype is for this call. The value ordering is defined by the GL tag in the VCF spec.
          3.14,
        ],
        "callsetName": "A String", # The name of the callset this variant call belongs to.
      },
    ],
    "created": "A String", # The date this variant was created, in milliseconds from the epoch.
    "referenceBases": "A String", # The reference bases for this variant. They start at the given position.
    "names": [ # Names for the variant, for example a RefSNP ID.
      "A String",
    ],
    "alternateBases": [ # The bases that appear instead of the reference bases.
      "A String",
    ],
    "position": "A String", # The position at which this variant occurs (1-based). This corresponds to the first base of the string of reference bases.
    "contig": "A String", # The contig on which this variant occurs. (such as 'chr20' or 'X')
    "id": "A String", # The Google generated ID of the variant, immutable.
    "datasetId": "A String", # The ID of the dataset this variant belongs to.
  }</pre>
</div>

</body></html>